kong/tools/queue.lua

```diff
@@ -100,6 +100,96 @@ local function make_queue_key(name)
 end
 
 
+local function internal_remaining_capacity(self)
+  local remaining_entries = self.max_entries - self:count()
+  local max_bytes = self.max_bytes
+
+  -- we enqueue entries one by one,
+  -- so it is impossible to have a negative value
+  assert(remaining_entries >= 0, "queue should not be over capacity")
+
+  if not max_bytes then
+    return remaining_entries
+  end
+
+  local remaining_bytes = max_bytes - self.bytes_queued
+
+  -- we check remaining_bytes before enqueueing an entry,
+  -- so it is impossible to have a negative value
+  assert(remaining_bytes >= 0, "queue should not be over capacity")
+
+  return remaining_entries, remaining_bytes
+end
+
+
+local function internal_is_reaching_max_entries(self)
+  -- `()` is used to get the first return value only
+  return (internal_remaining_capacity(self)) == 0
+end
+
+
+local function internal_will_exceed_max_entries(self)
+   -- `()` is used to get the first return value only
+  return (internal_remaining_capacity(self)) - 1 < 0
```

Comment:
`(internal_remaining_capacity(self))  < 1` may be easy to understand?

Reply:
I think both two representations are fine.